src/rust/engine/pantsd/src/lib.rs

```diff
@@ -35,11 +34,35 @@ use std::{fmt, fs};
 
 use libc::pid_t;
 use log::debug;
-use options::{option_id, BuildRoot, OptionId, OptionType};
+use options::{option_id, BuildRoot, OptionId, OptionParser, OptionType};
 use sha2::digest::Update;
 use sha2::{Digest, Sha256};
 use sysinfo::{ProcessExt, ProcessStatus, System, SystemExt};
 
+pub struct ConnectionSettings {
+  pub port: u16,
+  pub timeout_limit: f64,
+  pub dynamic_ui: bool,
+}
+
+impl ConnectionSettings {
+  pub fn new(port: u16) -> ConnectionSettings {
+    ConnectionSettings {
+      port,
+      timeout_limit: 60.0,
+      dynamic_ui: true,
+    }
+  }
+}
+
+fn to_hex(bytes: &[u8]) -> String {
+  let mut hex = String::with_capacity(bytes.len());
```

Comment:
(If we're keeping this function, this should be the following, or else it reallocs, because each byte becomes two once encoded:)

```suggestion
  let mut hex = String::with_capacity(bytes.len() * 2);
```